### ISSUE TITLE
[Mac OS] clang: error: linker command failed 

### ISSUE THREAD

**Issue by Humoon** [google-research/football#280]
Hi, I'm trying install boost@1.72.0 with source code, but when I successfully built all the libraries and installed boost, the following error occured. 
Here is my log:
```
  -- The C compiler identification is AppleClang 12.0.5.12050022
  -- The CXX compiler identification is AppleClang 12.0.5.12050022
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/gcc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/clang - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Found OpenGL: /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/OpenGL.framework
  -- Found SDL2_image: /usr/local/lib/libSDL2_image.dylib
  -- Found SDL2_ttf: /usr/local/lib/libSDL2_ttf.dylib
  -- Found SDL2_gfx: /usr/local/lib/libSDL2_gfx.dylib (found version "1.0.4")
  -- Found PythonInterp: /Users/huyueyue/Work/Code/football-env/bin/python (found version "3.7.11")
  Using Python: 3.7.11
  -- Could NOT find Boost: missing: python3-py37 (found /usr/local/lib/cmake/Boost-1.72.0/BoostConfig.cmake (found version "1.72.0"))
  python3-py37 not found. Trying other names.
  -- Could NOT find Boost: missing: python-py37 (found /usr/local/lib/cmake/Boost-1.72.0/BoostConfig.cmake (found version "1.72.0"))
  python-py37 not found. Trying other names.
  -- Found Boost: /usr/local/lib/cmake/Boost-1.72.0/BoostConfig.cmake (found version "1.72.0") found components: python37
  Using python_boost: python37
  -- Found Boost: /usr/local/lib/cmake/Boost-1.72.0/BoostConfig.cmake (found version "1.72.0") found components: thread system filesystem python37
  -- Configuring done
  -- Generating done
  -- Build files have been written to: /private/var/folders/ft/7v6gvd4n723cn8tyschnllxc0000gn/T/pip-req-build-7hj0sldg/third_party/gfootball_engine
  [  2%] Building CXX object CMakeFiles/baselib.dir/src/base/sdl_surface.cpp.o
  [  3%] Building CXX object CMakeFiles/utilslib.dir/src/utils/orbitcamera.cpp.o
  [  5%] Building CXX object CMakeFiles/scenelib.dir/src/scene/objectfactory.cpp.o
  [  4%] Building CXX object CMakeFiles/typeslib.dir/src/types/spatial.cpp.o
  [  5%] Building CXX object CMakeFiles/loaderslib.dir/src/loaders/imageloader.cpp.o
  [  7%] Building CXX object CMakeFiles/datalib.dir/src/data/matchdata.cpp.o
  ...
  [ 96%] Building CXX object CMakeFiles/game.dir/src/defines.cpp.o
  [ 97%] Building CXX object CMakeFiles/game.dir/ai.cpp.o
  [ 98%] Building CXX object CMakeFiles/game.dir/src/game_env.cpp.o
  [ 99%] Building CXX object CMakeFiles/game.dir/src/ai/ai_keyboard.cpp.o
  [100%] Linking CXX shared library libgame.dylib
  Undefined symbols for architecture x86_64:
    "_PyBool_FromLong", referenced from:
        boost::python::objects::caller_py_function_impl<boost::python::detail::caller<bool (*)(std::__1::vector<float, std::__1::allocator<float> >&, _object*), boost::python::default_call_policies, boost::mpl::vector3<bool, std::__1::vector<float, std::__1::allocator<float> >&, _object*> > >::operator()(_object*, _object*) in ai.cpp.o
        boost::python::objects::caller_py_function_impl<boost::python::detail::caller<bool (*)(std::__1::vector<int, std::__1::allocator<int> >&, _object*), boost::python::default_call_policies, boost::mpl::vector3<bool, std::__1::vector<int, std::__1::allocator<int> >&, _object*> > >::operator()(_object*, _object*) in ai.cpp.o
        boost::python::objects::caller_py_function_impl<boost::python::detail::caller<bool (*)(std::__1::vector<PlayerInfo, std::__1::allocator<PlayerInfo> >&, _object*), boost::python::default_call_policies, boost::mpl::vector3<bool, std::__1::vector<PlayerInfo, std::__1::allocator<PlayerInfo> >&, _object*> > >::operator()(_object*, _object*) in ai.cpp.o
        boost::python::objects::caller_py_function_impl<boost::python::detail::caller<bool (*)(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&, _object*), boost::python::default_call_policies, boost::mpl::vector3<bool, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&, _object*> > >::operator()(_object*, _object*) in ai.cpp.o
        boost::python::objects::caller_py_function_impl<boost::python::detail::caller<boost::python::detail::member<bool, PlayerInfo>, boost::python::return_value_policy<boost::python::return_by_value, boost::python::default_call_policies>, boost::mpl::vector2<bool&, PlayerInfo&> > >::operator()(_object*, _object*) in ai.cpp.o
```

The error message is:
```
  ld: symbol(s) not found for architecture x86_64
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  make[2]: *** [libgame.dylib] Error 1
  make[1]: *** [CMakeFiles/game.dir/all] Error 2
  make: *** [all] Error 2
  gfootball/build_game_engine.sh: line 27: pushd: third_party/gfootball_engine: No such file or directory
  error: Google Research Football compilation failed
error
  ERROR: Failed building wheel for gfootball
  Running setup.py clean for gfootball
  Running command /Users/huyueyue/Work/Code/football-env/bin/python3 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/ft/7v6gvd4n723cn8tyschnllxc0000gn/T/pip-req-build-7hj0sldg/setup.py'"'"'; __file__='"'"'/private/var/folders/ft/7v6gvd4n723cn8tyschnllxc0000gn/T/pip-req-build-7hj0sldg/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' clean --all
  running clean
  removing 'build/lib.macosx-11-x86_64-3.7' (and everything under it)
  'build/bdist.macosx-11-x86_64' does not exist -- can't clean it
  'build/scripts-3.7' does not exist -- can't clean it
  removing 'build'
Failed to build gfootball
Installing collected packages: gfootball
```
Thanks

**Comment by vi3itor** [comment]
It's hard to guide you, because you decided to use your own set up and compile boost from sources. Engine is compiled but linking step fails.

Do you have to use Python 3.7? If otherwise, you can install all dependencies with `brew` (including python 3.9 and boost-python39) and it should work out of the box.

**Comment by Humoon** [comment]
Well, I successfully installed the football environment for Python 3.9 with `brew`, but since my TensorFlow version is specified as 1.15.0, unfortunately this is not available in Python 3.9.  Thanks for your reply. I will find other solutions :)

**Comment by zixianma** [comment]
> Well, I successfully installed the football environment for Python 3.9 with `brew`, but since my TensorFlow version is specified as 1.15.0, unfortunately this is not available in Python 3.9. Thanks for your reply. I will find other solutions :)

@Humoon Hi, did you find out a solution? I'm experiencing the same issue even with python3.9. Thanks in advance!

**Comment by Humoon** [comment]
I think I have forgot how to solve it, but curruntly my environment with python3.8, boost-python3@1.72.0 and tensorflow2.3 which works well now.
